x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalOptimizerContext.java

```diff
@@ -35,7 +35,7 @@ public boolean equals(Object obj) {
         if (obj == this) return true;
         if (obj == null || obj.getClass() != this.getClass()) return false;
         var that = (LogicalOptimizerContext) obj;
-        return Objects.equals(this.configuration, that.configuration);
+        return this.configuration.equals(that.configuration) && this.foldCtx.equals(that.foldCtx);
```

Comment:
Should we update the `hashCode` implementation, too?

Reply:
Yes. Super important. You should block the merge for a mistake like that. Sorry.

It's not that it's *likely* to break anything, but it's sort of a bomb set up for someone years in the future.